indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -386,13 +388,13 @@ public void shutdown(final String taskId)
         log.info("Can't shutdown! No worker running task %s", taskId);
         return;
       }
-
+      URL url = null;
       try {
-        final URL url = makeWorkerURL(zkWorker.getWorker(), String.format("/task/%s/shutdown", taskId));
+        url = makeWorkerURL(zkWorker.getWorker(), String.format("/task/%s/shutdown", taskId));
         final StatusResponseHolder response = httpClient.go(
             new Request(HttpMethod.POST, url),
             RESPONSE_HANDLER
-        ).get();
+        ).get(config.getTaskShutdownLinkTimeout().getMillis(), TimeUnit.MILLISECONDS);
```

Comment:
Yes, will change when release of http-client is finished. (it is progressing right now)